src/meet-calendar.ts

```diff
@@ -0,0 +1,688 @@
+import { createRoom, generateRoomWithoutSeparator } from "./brave-talk";
+import $ from "jquery";
+
+const BASE_DOMAIN = "talk.brave.com";
+const BASE_URL = "https://" + BASE_DOMAIN + "/";
+const APP_NAME = "Brave Talk";
+
+//A text to be used when adding info to the location field.
+const LOCATION_TEXT = APP_NAME + " Meeting";
+
+/**
+ * The event page we will be updating.
+ */
+abstract class EventContainer {
+  // used to implement autocreate meetings, this is done after all
+  // the needed information is retrieved as numbers, upon calling update
+  public scheduleAutoCreateMeeting: boolean = false;
+
+  private containerElement: HTMLElement | undefined;
+  protected descriptionInstance: DescriptionWrapper | undefined;
+  protected locationInstance: LocationWrapper | undefined;
+  public meetingId: string | undefined;
+  /**
+   * @returns {EventContainer}
+   */
+  static getInstance(): EventContainer | undefined {
+    const body = document.querySelector("body");
+    if (body && body.dataset.viewfamily) return new G2Event(body);
+  }
+
+  /**
+   * The description of the event.
+   */
+  abstract get description(): DescriptionWrapper;
+
+  /**
+   * The button container where we will add the jitsi button.
+   */
+  abstract get buttonContainer(): JQuery<HTMLElement> | null;
+
+  /**
+   * The location of the event.
+   */
+  abstract get location(): LocationWrapper;
+
+  /**
+   * The container element of the event edit page.
+   */
+  get container(): HTMLElement | undefined {
+    return this.containerElement;
+  }
+
+  set container(c: HTMLElement | undefined) {
+    this.containerElement = c;
+  }
+
+  /**
+   * Main entry point of the event modifictaions.
+   */
+  abstract update(): void;
+
+  /**
+   * Checks for the button on current page
+   */
+  isButtonPresent() {
+    return document.getElementById("jitsi_button") !== null;
+  }
+
+  /**
+   * Clears instances.
+   */
+  reset() {
+    this.descriptionInstance = undefined;
+    this.locationInstance = undefined;
+  }
+
+  /**
+   * Updates meetingId, if there is meetingId set it, if not generate it.
+   */
+  updateMeetingId() {
```

Comment:
this logic is somewhat complicated and needs some tests

Reply:
logic significantly simplified and tests against stub html added